mlir/include/mlir/Dialect/Ptr/IR/MemorySpaceInterfaces.td

```diff
@@ -43,6 +43,7 @@ def MemorySpaceAttrInterface : AttrInterface<"MemorySpaceAttrInterface"> {
       /*args=*/        (ins "::mlir::Type":$type,
                             "::mlir::ptr::AtomicOrdering":$ordering,
                             "std::optional<int64_t>":$alignment,
+                            "std::optional<std::reference_wrapper<const ::mlir::DataLayout>>":$dataLayout,
```

Comment:
That seems uncommon in the codebase, the usual idiom for an optional thing would be to take a pointer:

```suggestion
                            "const ::mlir::DataLayout *":$dataLayout,
```